app/src/main/java/com/capstone/weatherapp/CityListFragment.kt

```diff
@@ -41,9 +45,40 @@ class CityListFragment : Fragment() {
         rvCities.layoutManager = LinearLayoutManager(context)
 
         viewModel.cityList.observe(viewLifecycleOwner) {
-            cityListAdapter.updateData(it)
+            when(it) {
+                is State.Success -> showSuccessState(it.result)
+                is State.Loading -> showLoadingState()
+                is State.Error -> showErrorState(it.message)
+            }
         }
         viewModel.getCityListFromRepo()
     }
 
+    fun showLoadingState() {
+        binding.rvCities.visibility = View.GONE
+        binding.stateLayout.visibility = View.VISIBLE
+        binding.detailsState.text = ""
+        Glide.with(this)
```

Comment:
We shouldn't need Glide for this image since it's static. Glide is only used for loading images from URL's. You can set the source of this image view in the xml file and then just set the visibility of this binding.detailsStateIcon instead